### PR TITLE
*: upgrade golangci-lint

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Upgrade golangci-lint

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### Release Note

 - No release note.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -89,13 +89,14 @@ check:
 static: export GO111MODULE=on
 static: prepare tools
 	@ # Not running vet and fmt through metalinter becauase it ends up looking at vendor
-	tools/bin/goimports -w -d -format-only -local $(BR_PKG) $$($(PACKAGE_DIRECTORIES)) 2>&1 | $(CHECKER)
+	tools/bin/gofumports -w -d -format-only -local $(BR_PKG) $$($(PACKAGE_DIRECTORIES)) 2>&1 | $(CHECKER)
```
Comment (lichunzhu) [inline Makefile:92]: Is `goimports` a subset of `gofumports`? I haven't found the relationship of them. Should we have them both?

**Review by lichunzhu — COMMENTED** [review]
LGTM

File: Makefile [hunk 2]
```diff
@@ -105,10 +106,15 @@ static: prepare tools
 	@#             nestif - PENDING REFACTORING
 	@#           goerr113 - it mistaken pingcap/errors with standard errors
 	@#                lll - pingcap/errors may need to write a long line
+	@#       paralleltest - no need to run test parallel
+	@#           nlreturn - no need to ensure a new line before continue or return
+	@#   exhaustivestruct - Protobuf structs have hidden fields, like "XXX_NoUnkeyedLiteral"
+	@#         exhaustive - no need to check exhaustiveness of enum switch statements
```
Comment (overvenus) [inline Makefile:112]:
```suggestion
	@#         exhaustive - no need to check exhaustiveness of enum switch statements
	@#              gosec - too many false positive
```

File: Makefile (hunk 1 above)
Comment (overvenus) [reply Makefile:92]: It's a stricter gofmt, also a drop-in replacement for goimports.

https://github.com/mvdan/gofumpt

File: pkg/storage/local.go [hunk 3]
```diff
@@ -12,6 +12,11 @@ import (
 	"github.com/pingcap/errors"
 )
 
+const (
+	localDirPerm  = os.FileMode(0o755)
```
Comment (lichunzhu) [inline pkg/storage/local.go:16]: Does hound mean this?
```suggestion
	localDirPerm  os.FileMode = 0755
```

**Review by kennytm — COMMENTED** [review]
LGTM after satisfying Hound 🐕

**Comment by overvenus** [comment]
I have no idea what's wrong with "0o644", it is suggested by gofumpt. 😥

**Comment by lichunzhu** [comment]
> I have no idea what's wrong with "0o644", it is suggested by gofumpt. 😥

Take a try on this?

```suggestion
	localDirPerm os.FileMode = 0o755
```

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**